recipes/caffe/build.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Setup CMake build location
+mkdir build
+cd build
+
+# Configure, build, test, and install.
+cmake -DCPU_ONLY=1 -DBLAS="open" -DCMAKE_INSTALL_PREFIX="${PREFIX}" ..
+make
+make runtest
+make install
+
+# Python installation is non-standard. So, we're fixing it.
+mv "${PREFIX}/python/caffe" "${SP_DIR}/"
+find "${PREFIX}/python/" -name "*.py" | xargs chmod +x
+find "${PREFIX}/python/" -name "*.py" | xargs -I {} cp {} "${PREFIX}/bin/"
```

Comment:
I don't think it is packaged in Debian or any other Linux distro. This is breaking edge research software for deep learning. Caffe is still doing release candidates. So, I think we are out of luck if we try to look for any kind of canonical installation via Linux package managers. Not to mention, most installation instructions involve a mixture of `conda`/`apt-get` installs and some sort of build in-place (so no proper install). Many are unofficial guides (e.g. I managed to get it to work like this). If anything, we are laying out the canonical install.

Caffe's install instructions actually suggest mucking with the `PYTHONPATH`, which feels pretty dirty to me (maybe to you too?). Especially, since putting the Python packages in `site-packages` and the binaries in `bin` feels so much cleaner and works the same.

I would definitely want to interact with upstream on a better way to do this Python install. Either by using `setup.py` or at a bare minimum changing the copy steps in their [CMakeLists.txt]( https://github.com/BVLC/caffe/blob/master/python/CMakeLists.txt#L24-L31 ).

Even though this is pretty rough and hacky, it doesn't seem nearly as bad as the alternatives. That all being said, by even attempting to package this in `conda` we are doing a major service to the Machine Learning community that has varying levels of proficiency in package management.